### ISSUE TITLE
defaultReloadStrategy fails on yarn test

### ISSUE THREAD

**Issue by minj** [mozilla/web-ext#2076]
### Is this a feature request or a bug?
Bug

### What is the current behavior?

https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=nodejs-web-ext&id=955d52d5922fea350448f3c9459fd423d48a4bea

```
      1) can still reload when user presses R after a reload error
[nodejs-web-ext/src/web-ext-build/src/extension-runners/index.js][info] Press R to reload (and Ctrl-C to quit)
[nodejs-web-ext/src/web-ext-build/src/extension-runners/index.js][debug] Reloading installed extensions on user request
[nodejs-web-ext/src/web-ext-build/src/extension-runners/index.js][debug] Reloading installed extensions on user request
[nodejs-web-ext/src/web-ext-build/src/extension-runners/index.js][info]
Exiting web-ext on user request

  1) util/extension-runners
       defaultReloadStrategy
         can still reload when user presses R after a reload error:
     Uncaught Error: fake reload error
      at FakeExtensionRunner.<anonymous> (tests/unit/test-extension-runners/test.extension-runners.js:539:37)
      at Function.invoke (node_modules/sinon/lib/sinon/proxy-invoke.js:47:47)
      at FakeExtensionRunner.spy (node_modules/sinon/lib/sinon/proxy.js:216:26)
      at Object.invoke (node_modules/sinon/lib/sinon/behavior.js:163:32)
      at FakeExtensionRunner.functionStub (node_modules/sinon/lib/sinon/stub.js:39:43)
      at Function.invoke (node_modules/sinon/lib/sinon/proxy-invoke.js:47:47)
      at FakeExtensionRunner.reloadAllExtensions (node_modules/sinon/lib/sinon/proxy.js:216:26)
      at nodejs-web-ext/src/web-ext-build/src/extension-runners/index.js:362:27


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### What is the expected or desired behavior?
Tests should pass

### Version information (for bug reports)

* **Firefox version**:
  * `/usr/bin/firefox -v`: Mozilla Firefox 83.0
  * `/usr/bin/firefox-developer-edition -v`: Mozilla Firefox 84.0b2
  * `/usr/bin/firefox-nightly -v`: Mozilla Firefox 84.0a1
  * **Your OS and version**: Manjaro linux x64
* Paste the output of these commands:

````
node --version && yarn --version && web-ext --version
v15.2.1
1.22.10
5.3.0
````

**Comment by rpl** [comment]
@minj I just tried to recreate the node_modules dir with yarn (both version 1 and version 2) locally on my system (Ubuntu 20.04) and both the test suites (unit and functional) do pass for me as they do with npm.

I did also tried to run them with node v15 (because we are not yet running our tests on v15 as part of our CI jobs) but that didn't trigger test failures neither.

If you can attach the yarn.lock file that is created for you on arch I may then try it to see if by using your yarn.lock I may be able to reproducible the same issue, unfortunately I don't have archlinux installation to quickly give the PKGBUILD file a try.

As a side note this repository does include a package-lock.json file and so using `npm ci` would provide a more predictable result (because the package version and node_modules tree would be the same on which we did run the test on the CI, because pinned in the package-lock.json file), on the contrary by using yarn there is a change to get a different result (especially when a sub-dependency doesn't pin an exact version of its own dependencies).

**Comment by Rob--W** [comment]
I was able to reproduce by downloading the PKGBUILD and running `makepkg -d`.

When I edit `PKGBUILD` and change 5.3.0 to the latest version, 5.4.0 and run `makepkg`, the issue does not occur any more.

I suggest to try again with the latest version. If the issue occurs again, add more info to the bug and we can re-open the issue.

**Comment by rpl** [comment]
ah, now I see why I wasn't able to reproduce it on master and this now makes sense to me.

This is a duplicate of #2053 (and the fix is in 5.4.0).

**Comment by minj** [comment]
5.4.0 confirmed working, thanks
